subprojects/core/src/main/java/org/gradle/api/internal/StartParameterInternal.java

```diff
@@ -156,7 +156,7 @@ public Option.Value<Boolean> getIsolatedProjects() {
 
     @Override
     public boolean isConfigurationCacheRequested() {
-        return configurationCache.get();
+        return configurationCache.get() || isolatedProjects.get();
```

Comment:
🤔 Why?

Reply:
We want the currently stable `isConfigurationCacheRequested` API to agree with the new API `buildFeatures.configurationCache.isRequested`. Since the use-case for the latter is accurate user-intent statistics gathering, it should reflect that requesting IP implies requesting CC. Otherwise, all the gathered statistics will be inaccurate for projects that enable only IP.

For more details, see this [section of the spec](https://docs.google.com/document/d/1xRB9Bps12YlzKalvb--JYD93IAbUNR_4ApcZqUmpj3g/edit#heading=h.74dn4h7tfdgz).

Reply:
As per the discussion in the spec we decided that the `requested` state will not be coupled between the features, and so this change was reverted.